src/main/java/org/junitpioneer/jupiter/cartesian/CartesianValueArgumentsProvider.java

```diff
@@ -14,25 +14,29 @@
 
 import java.lang.reflect.Array;
 import java.lang.reflect.Parameter;
+import java.util.Arrays;
 import java.util.List;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.support.AnnotationConsumer;
 import org.junit.platform.commons.PreconditionViolationException;
-import org.junit.platform.commons.support.AnnotationSupport;
 
 /**
- * This is a slightly modified copy of Jupiter's {@code ValueSourceArgumentsProvider},
- * except it does NOT support {@code @ParameterizedTest} and can consume a {@code Parameter}
- * instead of an annotation.
+ * This is a slightly modified copy of Jupiter's {@code ValueArgumentsProvider},
+ * except it does NOT support {@code @ParameterizedTest} and implements {@link CartesianArgumentsProvider}
+ * for use with {@code @CartesianTest}.
+ * <p>
+ * This class does not implement {@code ArgumentsProvider} since the Jupiter's {@code ValueSource} should be used for that.
```

Comment:
That last line is probably not needed.

Reply:
It was mostly there for an explanation for the maintainers why it does not implement it. However, I have removed it. Also removed it from `CartesianEnumArgumentsProvider`